ios/ReactNativeHealthkit.swift

```diff
@@ -994,25 +1037,39 @@ class ReactNativeHealthkit: RCTEventEmitter {
         let predicate = from != nil || to != nil ? HKQuery.predicateForSamples(withStart: from, end: to, options: [HKQueryOptions.strictEndDate, HKQueryOptions.strictStartDate]) : nil;
 
         let limit = limit == 0 ? HKObjectQueryNoLimit : limit;
-
-        let q = HKSampleQuery(sampleType: sampleType, predicate: predicate, limit: limit, sortDescriptors: [NSSortDescriptor(key: HKSampleSortIdentifierStartDate, ascending: ascending)]) { (query: HKSampleQuery, sample: [HKSample]?, error: Error?) in
-            guard let err = error else {
-                guard let samples = sample else {
-                    return resolve([]);
-                }
-                let arr: NSMutableArray = [];
-
-                for s in samples {
-                    if let sample = s as? HKQuantitySample {
-                        let serialized = self.serializeQuantitySample(sample: sample, unit: HKUnit.init(from: unitString))
-
-                        arr.add(serialized)
-                    }
-                }
-
-                return resolve(arr);
-            }
-            reject(GENERIC_ERROR, err.localizedDescription, err);
+    
+        let actualAnchor = anchor.isEmpty ? nil : base64StringToHKQueryAnchor(base64String: anchor)
+    
+        let q = HKAnchoredObjectQuery(
+          type: sampleType,
+          predicate: predicate,
+          anchor: actualAnchor,
+          limit: limit
+        ) { (
+          query: HKAnchoredObjectQuery,
+          s: [HKSample]?,
+          deletedSamples: [HKDeletedObject]?,
+          newAnchor: HKQueryAnchor?,
+          error: Error?
+        ) in
+          guard let err = error else {
+              guard let samples = s else {
+                  return resolve([]);
+              }
+
+            return resolve([
+              "samples": samples.map({ sample in
+                let serialized = self.serializeQuantitySample(sample: sample as! HKQuantitySample, unit: HKUnit.init(from: unitString))
+
+                return serialized
+              }) as Any,
+              "deletedSamples": deletedSamples?.map({ sample in
+                return self.serializeDeletedSample(sample: sample)
+              }) as Any,
+              "newAnchor": self.serializeAnchor(anchor: newAnchor!) as Any
```

Comment:
`newAnchor` can be `nil` in some cases like when user has not given permission for the identifier. In that case serialization fails. How about `newAnchor != nil ? ... : nil`?